### PR TITLE
Cleanup dependency to help with next.js migration

### PR DESCRIPTION
**Pull request by SamChou19815**

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -51,6 +51,7 @@
                 }
             }
         ],
+        "no-use-before-define": "off",
```
Comment (SamChou19815) [inline .eslintrc:54]: `@typescript-eslint/no-use-before-define` is already covering this. After version bump this one results in some weird errors.

File: .eslintrc [hunk 2]
```diff
@@ -65,7 +66,10 @@
         "react/state-in-constructor": "off",
         "import/extensions": "off",
         "indent": "off",
-        "@typescript-eslint/ban-ts-ignore": "warn",
+        "@typescript-eslint/ban-ts-comment": [
+            "error",
+            { "ts-expect-error": "allow-with-description" }
+        ],
```
Comment (SamChou19815) [inline .eslintrc:72]: The old one is removed in v3.

File: package.json [hunk 3]
```diff
@@ -42,29 +42,25 @@
     ]
   },
   "devDependencies": {
-    "@testing-library/jest-dom": "^5.5.0",
-    "@testing-library/react": "^9.3.2",
-    "@testing-library/user-event": "^7.1.2",
-    "@types/jest": "^25.2.1",
```
Comment (SamChou19815) [inline package.json:48]: looks like there are no tests 😢  so I removed them.

File: package.json [hunk 4]
```diff
@@ -42,29 +42,25 @@
...
+    "eslint": "^7.12.0",
+    "eslint-config-airbnb": "^18.2.0",
+    "eslint-plugin-import": "^2.22.1",
+    "eslint-plugin-jsx-a11y": "^6.3.1",
+    "eslint-plugin-react": "^7.21.5",
+    "eslint-plugin-react-hooks": "^4.2.0",
+    "stylelint": "^13.7.2",
     "stylelint-config-standard": "^20.0.0",
-    "typescript": "^3.8.3"
+    "typescript": "^4.0.3"
```
Comment (SamChou19815) [inline package.json:58]: bump bump

File: src/Landing/index.tsx [hunk 5]
```diff
@@ -1,7 +1,7 @@
 import React from 'react';
 import ChevronDown from './ChevronDown';
 import headshot from '../img/headshot.jpg';
-import * as styles from './index.module.css';
+import styles from './index.module.css';
```
Comment (SamChou19815) [inline src/Landing/index.tsx:4]: Not using the default export syntax is probably the reason why you added `src/declaration.d.ts`. Fixed it so no need to use extra declarations.

**Review by meganyin13 — APPROVED** [review]
thanks sam! ❤️
care

**Pull request merged**